spyder/widgets/sourcecode/base.py

```diff
@@ -886,61 +886,70 @@ def __move_line_or_selection(self, after_current_line=True):
         cursor = self.textCursor()
         cursor.beginEditBlock()
         start_pos, end_pos = self.__save_selection()
-        add_linesep = False
-        if to_text_string(cursor.selectedText()):

```

Comment:
The code under the section "Select text" is used to handle both cases, no need for a `if` condition anymore:
1) when no text is selected or if the selection only span over one line, current line is moved
2) when multiple lines are selected, all lines are moved